File: ASPNETCore/HowTo/WebApiTypescriptSample/src/Network/HttpParams.ts
import * as QueryString from 'querystring';
export class HttpParams {

    private Params: Map<string, string> = new Map<string, string>();

    public HttpParams() {
    }

    public Set(key: string, value: string): HttpParams {
        if (key == null || key.length == 0) return this;
        if (this.Params.has(key)) {
            this.Params.set(key, QueryString.escape(value));
        }
        else {
            this.Params.set(key, QueryString.escape(value));
        }
        return this;
    }
    public Append(key: string, value: string): HttpParams {
        if (key == null || key.length == 0) return this;
        if (this.Params.has(key)) {
            this.Params.set(key, this.Params.get(key) + `&${key}=${QueryString.escape(value)}`);
        }
        else {
            this.Params.set(key, QueryString.escape(value));
        }
        return this;
    }

    public ToString(): string {
        let prms: string = "";
        this.Params.forEach((value: string, key: string) => {
            prms += `${prms.length == 0 ? "" : "&"}${key}=${value}`;
        });
        return prms;
    }

}

File: ASPNETCore/HowTo/WebApiConsoleSample/src/Network/HttpParams.cs
using System;
using System.Collections.Generic;

namespace WebApiConsoleSample
{

    public class HttpParams
    {
        private Dictionary<string, string> Params = new Dictionary<string, string>();

        public HttpParams()
        {
        }

        public HttpParams Set(string key, string value)
        {
            if(key == null || key.Length == 0) return this;
            if (Params.ContainsKey(key))
            {
                Params[key] = Uri.EscapeDataString(value);
            }
            else
            {
                Params.Add(key, Uri.EscapeDataString(value));
            }
            return this;
        }
        public HttpParams Append(string key, string value)
        {
            if(key == null || key.Length == 0) return this;
            if (Params.ContainsKey(key))
            {
                Params[key] += String.Format("&{0}={1}", key, Uri.EscapeDataString(value));
            }
            else
            {
                Params.Add(key, Uri.EscapeDataString(value));
            }
            return this;
        }

        public override string ToString()
        {
            string prms = "";

            foreach (KeyValuePair<string, string> kvp in Params)
            {
                prms += String.Format("{0}{1}={2}", prms.Length == 0 ? "" : "&", kvp.Key, kvp.Value);
            }
            return prms;
        }
    }
}